core/domain/user_services_test.py

```diff
@@ -1318,6 +1603,242 @@ def test_record_user_started_state_translation_tutorial(self):
+
+        user_id = user_services.create_new_user(auth_id, user_email).user_id
+
+        error_msg = (
+            'User contributions model for user %s already exists.'
+            % user_id)
+
+        with self.assertRaisesRegex(Exception, error_msg):
+            user_services.create_user_contributions(
+                user_id,
+                created_exp_ids,
+                edited_exp_ids)
+
+    def test_create_user_contributions(self):
+        auth_id = 'someUser'
+        user_email = 'user@example.com'
+        created_exp_ids = ['exp1', 'exp2', 'exp3']
+        edited_exp_ids = ['exp2', 'exp3', 'exp4']
+
+        user_id = user_services.create_new_user(auth_id, user_email).user_id
+
+        for created_exp_id in created_exp_ids:
+            user_services.add_created_exploration_id(user_id, created_exp_id)
+        for edited_exp_id in edited_exp_ids:
+            user_services.add_edited_exploration_id(user_id, edited_exp_id)
+
+        contributions = user_services.get_user_contributions(user_id)
+
+        self.assertEqual(
+            created_exp_ids,
+            contributions.created_exploration_ids)
+
+        self.assertEqual(
+            edited_exp_ids,
+            contributions.edited_exploration_ids)
+
+    def test_update_user_contributions(self):
+        auth_id = 'someUser'
+        user_email = 'user@example.com'
+        created_exp_ids = ['exp1', 'exp2', 'exp3']
+        edited_exp_ids = ['exp2', 'exp3', 'exp4']
+
+        user_id = user_services.create_new_user(auth_id, user_email).user_id
+
+        user_services.update_user_contributions(
+            user_id,
+            created_exp_ids,
+            edited_exp_ids)
+
+        contributions = user_services.get_user_contributions(user_id)
+
+        self.assertEqual(
+            created_exp_ids,
+            contributions.created_exploration_ids)
+
+        self.assertEqual(
+            edited_exp_ids,
+            contributions.edited_exploration_ids)
+
+    def test_update_user_contributions_for_invalid_user_raises_error(self):
+        non_existent_user_id = 'id_x'
+        created_exp_ids = ['exp1', 'exp2', 'exp3']
+        edited_exp_ids = ['exp2', 'exp3', 'exp4']
+        error_msg = (
+            'User contributions model for user %s does not exist.'
+            % non_existent_user_id)
+
+        with self.assertRaisesRegex(Exception, error_msg):
+            user_services.update_user_contributions(
+            non_existent_user_id,
+            created_exp_ids,
+            edited_exp_ids)
+
+    def test_add_created_exploration_id(self):
+        auth_id = 'someUser'
+        user_email = 'user@example.com'
+        created_exp_ids = ['exp1']
+
+        user_id = user_services.create_new_user(auth_id, user_email).user_id
+
+        user_services.add_created_exploration_id(user_id, 'exp1')
+
+        contributions = user_services.get_user_contributions(user_id)
+
+        self.assertEqual(
+            created_exp_ids,
```

Comment:
Inline the expected value. If you want to make it its own variable, then "expected" should be somewhere in the name.

Reply:
Not done.

Reply:
Done. I inlined expected values.